appinventor/components/src/com/google/appinventor/components/runtime/WebViewer.java

```diff
@@ -483,6 +490,18 @@ public void Reload() {
           "This property is available only in the designer.")
   public void UsesLocation(boolean uses) {
     // We don't actually do anything here (the work is in the MockWebViewer)
+    this.usesLocation = uses;
+  }
+
+  /**
+   * Determine if the app uses location permissions.
+   * @return true if uses location
+   */
+  @SimpleProperty(userVisible = false,
+      description = "Whether or not to give the application permission to use the Javascript "
+          + "geolocation API. This property is available only in the designer.")
+  public boolean UsesLocation() {
```

Comment:
Implement conditional permission here.

Ref https://github.com/mit-cml/appinventor-sources/blob/080ec9208b453814fa24f39c6caf53a0eedec421/appinventor/components/src/com/google/appinventor/components/runtime/PhoneCall.java#L161

Reply:
@pavi2410 this is a getter method do I need to add conditional permission in getters as well?

Reply:
No, applying annotation on either of a setter or a getter is fine

Reply:
@pavi2410 ok,  so I just need to add usespermission annotation in either of getter or setter. The conditional requestnig of permission is already handled in EclairUtils file so won't have to add anything else in getter or setter method right? Thanks :-)

Reply:
Yes, just add the annotation. No more changes required